.NET/Microsoft.Recognizers.Definitions/English/DateTimeDefinitions.cs

```diff
@@ -167,9 +168,9 @@ public static class DateTimeDefinitions
 		public const string TokenBeforeTime = "at ";
 		public const string AMTimeRegex = @"(?<am>morning)";
 		public const string PMTimeRegex = @"\b(?<pm>afternoon|evening|night)\b";
-		public const string BeforeRegex = @"\b(before|in advance of|prior to|no later than|by|till|til|until)\b";
-		public const string AfterRegex = @"\b(after)\b";
-		public const string SinceRegex = @"\b(since)\b";
+		public const string BeforeRegex = @"\b(before|in advance of|prior to|(no later|earlier|sooner) than|ending (with|on)|by|till|til|until)\b";
+		public const string AfterRegex = @"\b(after(?!\s+or equal to)|(?<!no\s+)later than)\b";
+		public const string SinceRegex = @"\b(since|after or equal to|starting (from|on|with))\b";

```

Comment:
"shark attacks by year and whether or not they were fatal since 1840"
doesn't work

Reply:
I tested this, "since 1840" was extracted and parsed correctly.